### PR TITLE
Refactor SSL/TLS connection options

### PR DESCRIPTION
**Pull request by richsalz**
This makes the ordering follow the same as the man1 documentation.

Still to come, another PR that does editing.

This depends on #10159

### CODE REVIEW

File: doc/man3/SSL_CONF_cmd.pod [hunk 1]
```diff
@@ -9,222 +9,229 @@ SSL_CONF_cmd - send configuration command
...
 
-=item B<-comp>
+This sets the temporary curve used for ephemeral ECDH modes. Only used
+by servers.
 
-Enables support for SSL/TLS compression, same as clearing
-B<SSL_OP_NO_COMPRESSION>.
-This command was introduced in OpenSSL 1.1.0.
-As of OpenSSL 1.1.0, compression is off by default.
+The B<grups> argument is a curve name or the special value B<auto> which
```
Comment (FdaSilvaYY) [inline doc/man3/SSL_CONF_cmd.pod]: typo : B<grups> ?

Comment (richsalz) [reply doc/man3/SSL_CONF_cmd.pod]: updated commit pushed; thanks!

**Comment by richsalz** [comment]
Rebased since #10159 went in.  Also changed the title of this PR to be more accurate.

**Comment by richsalz** [comment]
rebased against master, fixed conflict pushed.

**Comment by richsalz** [comment]
I had to rebase because there was a conflict in s_server.pod.in  I also moved a few options around to group them for clarity.

**Comment by richsalz** [comment]
Rebased to fix conflict in s_server.pod.in

**Comment by richsalz** [comment]
Rebased to fix conflict in doc/perlvars.pm

**Comment by richsalz** [comment]
Rebased to pick up the fix in #10932; no other changes made.

**Comment by richsalz** [comment]
Rebased to fix a merge error in s_serrer.pod.in, duplicate lines; nothing else changed.

**Comment by richsalz** [comment]
Rebased and pushed, to fix trivial conflict in s_server.pod.in  No other changes made.

**Comment by richsalz** [comment]
ping for reviews, please.

**Comment by richsalz** [comment]
Thanks, @beldmit .  The label is outdated now. :)

**Comment by mspncp** [comment]
openssl-s_server.pod.in has a conflict which needs to be resolved:

```
<<<<<<< variant A
[B<-recv_max_early_data> I<int>]
>>>>>>> variant B
[B<-http_server_binmode>]
####### Ancestor
======= end
```

**Comment by richsalz** [comment]
I fixed the conflict.  In doing so, I moved the "binmode" option in the SYNOPSIS section.

**Comment by richsalz** [comment]
So is this ready for merge now?

**Comment by mspncp** [comment]
Do you want me to squash the two commits or should I keep them separate?

**Comment by mspncp** [comment]
Merged to master, thank you!

03e16083ff Fix "ts" no-XXX options, document two TLS options
8b3efb5302 Update the SSL/TLS connection options

**Comment by richsalz** [comment]
Thanks!

**Pull request closed without merging**